File: 프로그래머스/1/42862. 체육복/체육복.java
class Solution {
    static int count = 0;
    static int arr[];
    public int solution(int n, int[] lost, int[] reserve) {
        arr = new int[n];
        for(int i=0; i<n; i++){
            arr[i] = 1;
        }
        for(int i=0; i<lost.length; i++){
            arr[lost[i]-1] --;
        }
        for(int i=0; i<reserve.length; i++){
            arr[reserve[i]-1] ++;
        }
        for(int i=0; i<arr.length; i++){
            if(arr[i] == 0){
                if(i>0 && arr[i-1] == 2){
                    arr[i-1] --;
                    arr[i] ++;
                }
                else if(i<arr.length-1 && arr[i+1] == 2){
                    arr[i+1] --;
                    arr[i] ++;
                }
            }
        }
        for(int i=0; i<arr.length; i++){
            if(arr[i] > 0){
                count++;
            }
        }
        return count;
    }
}

File: 프로그래머스/1/42862. 체육복/체육복.cpp
#include <string>
#include <vector>
#include <algorithm>
using namespace std;

int solution(int n, vector<int> lost, vector<int> reserve) {
    vector<int> arr(n+1, 0);
    sort(lost.begin(), lost.end());
    sort(reserve.begin(), reserve.end());
    for(int i=0; i<lost.size(); i++){
        arr[lost[i]]--;
    }
    for(int i=0; i<reserve.size(); i++){
        arr[reserve[i]]++;
    }
    for(int i=0; i<reserve.size(); i++){
        if(arr[reserve[i]] == 1){
            if(arr[reserve[i] - 1] == -1){
                arr[reserve[i] - 1]++;
            }
            else if(arr[reserve[i] + 1] == -1){
                arr[reserve[i] + 1]++;
            }
        }
    }
    int answer = 0;
    for(int i=1; i<=n; i++){
        if(arr[i] != -1){
            answer ++;
        }
    }
    return answer;
}